shared/bots/__init__.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from typing import List, Optional
+
+from shared.bots.github_apps import get_github_app_info_for_owner
+from shared.bots.owner_bots import get_owner_appropriate_bot_token
+from shared.bots.public_bots import get_token_type_mapping
+from shared.bots.repo_bots import get_repo_appropriate_bot_token
+from shared.bots.types import (
+    AdapterAuthInformation,
+)
+from shared.django_apps.codecov_auth.models import (
+    GITHUB_APP_INSTALLATION_DEFAULT_NAME,
+    Owner,
+    Service,
+)
+from shared.django_apps.core.models import Repository
+from shared.typings.torngit import GithubInstallationInfo
+
+log = logging.getLogger(__name__)
+
+
+def get_adapter_auth_information(
+    owner: Owner,
```

Comment:
If this is required to be a Django model, will this code be able to be used by the worker?

If not (I suspect not at this time) is there a plan to move the worker to use this code? (instead of having this all repeated in the worker as well)

Reply:
The main differences are 
a) repo.owner -> repo.author
b) owner.github_app_installations -> owner.github_app_installations.all()

For a), we could add the author relationship and it should be fine, although I'm not sure if sqlalchemy would complain to have 2 keys with the same relationship. But b) is a django specific workaround so that one would be a bit harder to do. 

We could make it work by passing the django Owner/Repository to this function every time it's used, the DB transactions could get a bit weird, but this logic doesn't seem to persist anything, it's just gets, so it could work

Reply:
If there is the _intention_ of using only the Django models (including in the worker) for the medium-future I think it's fine to force the interface using Django models.

Otherwise we might think about establishing a common interface (through getter functions) in both models.